commit e01eb40e4242647cfd61416192040806cde46a1e (refs/changes/46/646/2)
Author: Todd Whiteman <todd.whiteman@joyent.com>
Date:   2016-10-11T10:41:52-07:00 (3 years ago)
    
    DOCKER-663 Support Amazon ECR Registry
    Reviewed by: Trent Mick <trentm@gmail.com>

diff --git a/CHANGES.md b/CHANGES.md
index 9761923..0b4c881 100644
--- a/CHANGES.md
+++ b/CHANGES.md
@@ -1,5 +1,9 @@
 # IMGAPI changelog
 
+## 3.1.1
+
+- DOCKER-663 Support Amazon ECR Registry
+
 ## 3.1.0
 
 - Updates to support using node v4 (IMGAPI-587).
diff --git a/lib/images.js b/lib/images.js
index 66b7845..179f24e 100644
--- a/lib/images.js
+++ b/lib/images.js
@@ -3389,7 +3389,8 @@ function apiAdminImportDockerImage(req, res, callback) {
             }
             var ref = rat.tag || rat.digest;
             // TODO: Can we send in pingRes, from earlier 'supportsV2' call?
-            ctx.regClientV2.getManifest({ref: ref}, function (err, man, res_) {
+            ctx.regClientV2.getManifest({ref: ref},
+                    function (err, man, res_, manifestStr) {
                 if (err) {
                     if (err.statusCode === 404 || err.statusCode === 401) {
                         /*
@@ -3412,6 +3413,12 @@ function apiAdminImportDockerImage(req, res, callback) {
                         'v2.getManifest found');
                     ctx.manifestV2 = man;
                     ctx.digestV2 = res_.headers['docker-content-digest'];
+                    if (!ctx.digestV2) {
+                        // Some registries (looking at you Amazon ECR) do not
+                        // provide the docker-content-digest header in the
+                        // response, so we have to calculate it.
+                        ctx.digestV2 = drc.digestFromManifestStr(manifestStr);
+                    }
                     ctx.regV = 2;   // The signal we'll be using v2.
                     next();
                 }
diff --git a/package.json b/package.json
index 280820c..c3d91f6 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "imgapi",
   "description": "Image API to manage images for SDC 7",
-  "version": "3.0.0",
+  "version": "3.1.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
@@ -10,7 +10,7 @@
     "bunyan": "1.8.1",
     "cmdln": "3.2.1",
     "dashdash": "1.10.0",
-    "docker-registry-client": "3.2.0",
+    "docker-registry-client": "3.2.1",
     "effluent-logger": "git+https://github.com/joshwilsdon/effluent-logger.git#d662f161a07f94045ad2afb45442931511c40e51",
     "expiring-lru-cache": "2.1.0",
     "extsprintf": "1.2.0",
